integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java

```diff
@@ -154,8 +154,16 @@ private static CarbonTable getCarbonTable(Configuration configuration, String pa
     } else {
       carbonInputFormat = new CarbonFileInputFormat<>();
     }
-    List<org.apache.hadoop.mapreduce.InputSplit> splitList =
-        carbonInputFormat.getSplits(jobContext);
+    List<org.apache.hadoop.mapreduce.InputSplit> splitList;
+    try {
+      splitList = carbonInputFormat.getSplits(jobContext);
+    } catch (IOException ex) {
+      if (ex.getMessage().contains("No Index files are present in the table location :")) {
```

Comment:
I see that hive supports only non-transactional table support. Non trascational tables are often created by the path provided by user.  so, when query if the path doesn't have carbon files. It is must to throw exception. 

If external table is created on wrong path. Now query gives 0 rows and user will not know what went wrong.

Reply:
so test for external table and don't skip exception for external table

Reply:
The external table is not getting queried from this getSplits() of this file.